### PR TITLE
coalib: Fix affected_code placeholder render

### PR DESCRIPTION
**Pull request by vijeth-aradhya**
### Checklist

- [ ] I have rebased properly. Please see [our tutorial on rebasing](http://coala.io/git#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.io/commit) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://coala.io/commit#shortlog).
- [x] I have followed the [guidelines for the commit body](http://coala.io/commit#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
- [x] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### Reviewers

@Makman2 
@Abhisek-

### CODE REVIEW

File: coalib/output/ConsoleInteraction.py [hunk 1]
```diff
@@ -304,25 +305,28 @@ def print_results_formatted(log_printer,
 
     for result in result_list:
         severity_str = RESULT_SEVERITY.__str__(result.severity)
+        args = result.__dict__
```
Comment (Makman2) [inline coalib/output/ConsoleInteraction.py]: you could also use `dir` or `vars`, they are maybe better as these are built-ins^^

Comment (vijeth-aradhya) [inline coalib/output/ConsoleInteraction.py]: Also I don't why gitmate-bot is giving error on quote marks still :sweat_smile: If you see my forked repo, then you can see that I have made the required changes :smile:

Comment (Makman2) [inline coalib/output/ConsoleInteraction.py]: Did you repush properly? On your fork it's not appearing to be fixed either :)

Comment (vijeth-aradhya) [inline coalib/output/ConsoleInteraction.py]: Oh it's seeing the older commit :sweat_smile: I'll fix it :+1:

**Comment by Makman2** [comment]
you fix the same issue twice, only one of them can be the actual fix, and it's likely the 2nd commit :)

File: coalib/output/ConsoleInteraction.py [hunk 2]
```diff
@@ -289,7 +289,8 @@ def print_results_formatted(log_printer,
     default_format = ('id:{id}:origin:{origin}:file:{file}:line:{line}:'
                       'column:{column}:end_line:{end_line}:end_column:'
                       '{end_column}:severity:{severity}:severity_str:'
-                      '{severity_str}:message:{message}')
+                      '{severity_str}:message:{message}:affected_code:'
+                      '{affected_code}')
```
Comment (Makman2) [inline coalib/output/ConsoleInteraction.py]: you change the default format, this would
1. belong to a different commit
2. need discussion/issue or at least a good reason why you change it

File: coalib/output/ConsoleInteraction.py [hunk 3]
```diff
@@ -304,25 +305,28 @@ def print_results_formatted(log_printer,
 
     for result in result_list:
         severity_str = RESULT_SEVERITY.__str__(result.severity)
+        args = vars(result)
         try:
             if len(result.affected_code) == 0:
+                args['affected_code'] = None
```
Comment (Makman2) [inline coalib/output/ConsoleInteraction.py]: Isn't it `None` by default?

File: coalib/output/ConsoleInteraction.py [hunk 4]
```diff
@@ -304,25 +305,28 @@ def print_results_formatted(log_printer,
...
                                         end_line=None,
                                         column=None,
                                         end_column=None,
                                         severity_str=severity_str,
-                                        **result.__dict__))
+                                        **args))
                 continue
 
             for range in result.affected_code:
+                args['affected_code'] = range
```
Comment (Makman2) [inline coalib/output/ConsoleInteraction.py]: this looks wrong, if there's a `SourceRange`, why do you reassign it to the `range` built-in function object? It think you should just delete this line^^

**Comment by Makman2** [comment]
And btw: For the commit body you are allowed to use 72 chars, so you don't have to wrap so often :)

File: coalib/results/SourceRange.py [hunk 5]
```diff
@@ -126,6 +126,27 @@ def __json__(self, use_relpath=False):
             _dict['file'] = relpath(_dict['file'])
         return _dict
 
+    def __str__(self):
+        """
+        Default __str__ representation -
+            Filename: FROM start_line:start_coloumn TO end_line:end_coloumn
```
Comment (Makman2) [inline coalib/results/SourceRange.py]: it's `column` not `coloumn` :)

File: coalib/results/SourceRange.py [hunk 6]
```diff
@@ -126,6 +126,27 @@ def __json__(self, use_relpath=False):
...
+        If whole line is affected -
+            Filename: FROM start_line TO end_line
+        If whole file is affected -
+            Filename
```
Comment (Makman2) [inline coalib/results/SourceRange.py]: Would suggest you reformat this a bit to this to make it look prettier in RST:
```
        Default representation::

            filename: FROM start_line:start_column TO end_line:end_column

        If whole line is affected::

            filename: FROM start_line TO end_line

        If whole file is affected::

            filename
```

File: coalib/results/SourceRange.py [hunk 7]
```diff
@@ -126,6 +126,27 @@ def __json__(self, use_relpath=False):
...
+            return self.start.file
+
+        from_str = ': FROM ' + str(self.start.line)
+        to_str = ' TO ' + str(self.end.line)
+
+        if self.start.column is None and self.end.column is None:
+            return self.start.file + from_str + to_str
+
+        return (self.start.file + from_str + ':' + str(self.start.line) +
+                to_str + ':' + str(self.end.line))
```
Comment (Makman2) [inline coalib/results/SourceRange.py]: 1. you need to write a testcase for this in `SourceRangeTest`
2. I think it's a bit simpler to read if you do this:
   ```python
   if line is None or column is None:
        format_str = '...'
    elif other_condition:
        # choose different format
    elif another_condition:
        ...
    return format_str.format(...)
   ```

**Comment by Makman2** [comment]
It seems there were also some other test cases which already tested the format string a bit, you need to modify those properly too :+1:

File: coalib/output/ConsoleInteraction.py (hunk 4 above)
Comment (vijeth-aradhya) [inline coalib/output/ConsoleInteraction.py]: `affected_code` is a tuple of `SourceRange`s right? Not a `SourceRange`

Comment (Abhisek-) [inline coalib/output/ConsoleInteraction.py]: line number 321(which is already in master branch) looks wrong because somebody is assigning the value to the built-in function `range`. I'll raise an issue by suggesting changes for that to something like `sourcerange` instead.

File: coalib/output/ConsoleInteraction.py [hunk 8]
```diff
@@ -304,25 +305,28 @@ def print_results_formatted(log_printer,
...
             if len(result.affected_code) == 0:
+                args['affected_code'] = None
                 print(format_str.format(file=None,
                                         line=None,
                                         end_line=None,
                                         column=None,
                                         end_column=None,
                                         severity_str=severity_str,
-                                        **result.__dict__))
+                                        **args))
```
Comment (Abhisek-) [inline coalib/output/ConsoleInteraction.py]: `args`(Considering you have not changed the `args` variable which is passed to this enclosing function) will print the same information on every iteration of the for loop and of-course `args` may not contain what we want.

File: coalib/output/ConsoleInteraction.py [hunk 9]
```diff
@@ -304,25 +305,28 @@ def print_results_formatted(log_printer,
 
     for result in result_list:
         severity_str = RESULT_SEVERITY.__str__(result.severity)
+        args = vars(result)
```
Comment (Abhisek-) [inline coalib/output/ConsoleInteraction.py]: You are overwriting the `args` variable which is passed to this enclosing function. which is bad

File: coalib/output/ConsoleInteraction.py (hunk 8 above)
Comment (vijeth-aradhya) [inline coalib/output/ConsoleInteraction.py]: The reason I made another variable `args` which contains the same `dict` from `result.__dict__` is because `result.affected_code` is `tuple` of `SourceRange` objects, and each `SourceRange` object in that tuple gets printed once, i.e in each iteration. **All the other values remain the same, just like in the previous implementation**. Otherwise, I have not changed anything.

File: coalib/output/ConsoleInteraction.py (hunk 9 above)
Comment (vijeth-aradhya) [inline coalib/output/ConsoleInteraction.py]: I guess it's fine, if you see even the `severity_str` is getting overwritten for every iteration ^ Like I mentioned in the other comment, it _is_ required.

**Comment by vijeth-aradhya** [comment]
All the tests pass on my machine. I need to figure out why Circlecl and Travis-cl are failing.

**Comment by vijeth-aradhya** [comment]
@Nosferatul I rebased with the current `coala` repo, tested it and pushed it as well :sweat_smile: Didn't work

**Comment by Nosferatul** [comment]
I guess you should test the new method added

File: coalib/results/SourceRange.py [hunk 10]
```diff
@@ -126,6 +126,29 @@ def __json__(self, use_relpath=False):
...
+            format_str = '{file}: FROM {start_line} TO {end_line}'
+        else:
+            format_str = ('{file}: FROM {start_line}:{start_column} ' +
+                          'TO {end_line}:{end_column}')
+
+        return format_str.format(file=self.start.file,
+                                 start_line=self.start.line,
+                                 end_line=self.end.line,
+                                 start_column=self.start.column,
+                                 end_column=self.end.column)
```
Comment (Makman2) [inline coalib/results/SourceRange.py]: ah what you could also do to reduce code :)
```
format_str = '{0.start.file}: FROM {0.start.line} TO {self.end.line}'
# do this for all format strings, and at the end:
return format_str.format(self)
```

File: tests/results/SourceRangeTest.py [hunk 11]
```diff
@@ -86,6 +86,18 @@ def test_json(self):
...
+        compare = abspath('test_file') + ': FROM 1:1 TO 2:1'
+        self.assertEqual(uut, compare)
+
+        uut = str(SourceRange.from_values('test_file', 1, None, 2, None))
+        compare = abspath('test_file') + ': FROM 1 TO 2'
+        self.assertEqual(uut, compare)
+
+        uut = str(SourceRange.from_values('test_file', None, None, None, None))
+        self.assertEqual(uut, abspath('test_file'))
+
```
Comment (Makman2) [inline tests/results/SourceRangeTest.py]: please squash the test with the commit that introduces `__str__`.

File: coalib/output/ConsoleInteraction.py [hunk 12]
```diff
@@ -312,17 +313,18 @@ def print_results_formatted(log_printer,
                                         column=None,
                                         end_column=None,
                                         severity_str=severity_str,
-                                        **result.__dict__))
+                                        **format_args))
                 continue
 
             for range in result.affected_code:
+                format_args['affected_code'] = range
```
Comment (Makman2) [inline coalib/output/ConsoleInteraction.py]: ahhh omg sorry... that's the problem when built-in names are reused :3
Yes this makes totally sense :+1: :+1:
sorry again :D

File: coalib/output/ConsoleInteraction.py [hunk 13]
```diff
@@ -304,6 +304,7 @@ def print_results_formatted(log_printer,
 
     for result in result_list:
         severity_str = RESULT_SEVERITY.__str__(result.severity)
+        format_args = vars(result)
         try:
             if len(result.affected_code) == 0:
```
Comment (Makman2) [inline coalib/output/ConsoleInteraction.py]: shouldn't you set `affected_code` to `None` or `Unknown` or something? I think you did that before, sorry for not understanding that earlier^^

File: coalib/results/SourceRange.py [hunk 14]
```diff
@@ -126,6 +126,29 @@ def __json__(self, use_relpath=False):
             _dict['file'] = relpath(_dict['file'])
```
Comment (Makman2) [inline coalib/results/SourceRange.py]: commit message: Adds -> Add

File: coalib/results/SourceRange.py [hunk 15]
```diff
@@ -126,6 +126,29 @@ def __json__(self, use_relpath=False):
...
         return _dict
 
+    def __str__(self):
+        """
+        Default representation::
+            filename: FROM start_line:start_column TO end_line:end_column
+        If whole line is affected::
+            filename: FROM start_line TO end_line
+        If whole file is affected::
+            filename
```
Comment (Makman2) [inline coalib/results/SourceRange.py]: I've thought about those colons again, and I get permanently confused when seeing them, as normally they already indicate (line)ranges, e.g. `3:4` means for me line 3 to 4. I think we should change the format str either to
```
FROM L3C3 TO L10C7
# or
L3C3:L10C7
# or with more spaces if it's too unreadable
L3 C3 : L10 C7
```

Comment (Makman2) [inline coalib/results/SourceRange.py]: Wha tdo you think @vijeth-aradhya ?

Comment (Abhisek-) [inline coalib/results/SourceRange.py]: If you ask me, I would say the last one is the best :)

File: tests/results/SourceRangeTest.py (hunk 11 above)
Comment (vijeth-aradhya) [inline tests/results/SourceRangeTest.py]: Alright I will :smile: But, I was thinking one commit for each file would be neat?

File: coalib/results/SourceRange.py (hunk 15 above)
Comment (vijeth-aradhya) [inline coalib/results/SourceRange.py]: Yes this is more readable and better to understand :smile: 
`L3 C3 : L10 C7` looks good I guess

File: tests/results/SourceRangeTest.py (hunk 11 above)
Comment (Abhisek-) [inline tests/results/SourceRangeTest.py]: All of your changes serves one purpose, making multiple commits doesn't make sense.

File: coalib/results/SourceRange.py [hunk 16]
```diff
@@ -126,6 +126,27 @@ def __json__(self, use_relpath=False):
             _dict['file'] = relpath(_dict['file'])
         return _dict
 
+    def __str__(self):
+        """
+        Default representation::
```
Comment (sils) [inline coalib/results/SourceRange.py]: add newline before and after code samples please, also can you make this a doctest?

**Comment by sils** [comment]
nice, I like the direction of this :) 'til now you aren't really supposed to use affected code at all are you? You gotta also add a commit to DefaultArgParser adding the info to the help.

File: coalib/results/SourceRange.py (hunk 16 above)
Comment (vijeth-aradhya) [inline coalib/results/SourceRange.py]: Yeah sure, first I'll let the representations as it is right now, and then provide three doctests corresponding to each representation?

Comment (sils) [inline coalib/results/SourceRange.py]: yep

File: coalib/results/SourceRange.py [hunk 17]
```diff
@@ -126,6 +126,42 @@ def __json__(self, use_relpath=False):
             _dict['file'] = relpath(_dict['file'])
         return _dict
 
+    def __str__(self):
+        """
+        Creates a string representation of the SourceRange object
+
+        1. Default representation::
```
Comment (sils) [inline coalib/results/SourceRange.py]: rather explain the doctests than doubling the whole thing, huh?

File: tests/results/SourceRangeTest.py [hunk 18]
```diff
@@ -86,6 +86,18 @@ def test_json(self):
...
+        uut = str(SourceRange.from_values('test_file', 1, 1, 2, 1))
+        compare = abspath('test_file') + ': L1 C1 : L2 C1'
+        self.assertEqual(uut, compare)
+
+        uut = str(SourceRange.from_values('test_file', 1, None, 2, None))
+        compare = abspath('test_file') + ': L1 : L2'
+        self.assertEqual(uut, compare)
+
+        uut = str(SourceRange.from_values('test_file', None, None, None, None))
+        self.assertEqual(uut, abspath('test_file'))
```
Comment (Makman2) [inline tests/results/SourceRangeTest.py]: those testcases are the same from the doctest, so you can actually remove `test_str` :)

File: coalib/results/SourceRange.py [hunk 19]
```diff
@@ -126,6 +126,51 @@ def __json__(self, use_relpath=False):
...
+        >>> case1 == example_str1
+        True
+
+        If the whole file is affected, then just the filename is shown.
+
+        >>> case2 = abspath('test_file') + ': L1 : L2'
+        >>> example_str2 = str(SourceRange.from_values('test_file', 1,
+        ... None, 2, None))
+        >>> case2 == example_str2
+        True
```
Comment (Makman2) [inline coalib/results/SourceRange.py]: you could also try to use ` # +ELLIPSIS`, so you can do this:
```
>>> str(SourceRange.from_values('test_file', 1, None, 2, None) # +ELLIPSIS
...test_file: L1 : L2
```
then you don't need the `abspath` :)

**Comment by sils** [comment]
you don't need the `+ELLIPSIS` for this to work, it's active by default

Sincerely,

Lasse Schuirmann

lasse@schuirmann.net
http://coala.io/ - http://viperdev.io/ - http://gitmate.com/

File: coalib/results/SourceRange.py (hunk 19 above)
Comment (vijeth-aradhya) [inline coalib/results/SourceRange.py]: I tried 
```
>>> str(SourceRange.from_values('test_file', None, None, None, None)) # +ELLIPSIS
test_file
```
It doesn't work

File: tests/results/SourceRangeTest.py (hunk 18 above)
Comment (vijeth-aradhya) [inline tests/results/SourceRangeTest.py]: Is it better to edit the `test_str` method by giving different values or remove the method itself?

File: coalib/results/SourceRange.py (hunk 19 above)
Comment (Makman2) [inline coalib/results/SourceRange.py]: no you need the `...` which indicates the same as `.*` for regexes :)
so -> `...test_file`

File: tests/results/SourceRangeTest.py (hunk 18 above)
Comment (Makman2) [inline tests/results/SourceRangeTest.py]: If you have a different corner case you want to test, then yes, but I think your current doctests do suffice^^

Comment (vijeth-aradhya) [inline tests/results/SourceRangeTest.py]: What if we have starting column number but ending column number as None? I could give those kind of testcases.

File: coalib/results/SourceRange.py (hunk 19 above)
Comment (vijeth-aradhya) [inline coalib/results/SourceRange.py]: Hmm I still get this error when I run the tests for the above comment of mine without the space after `...`
```
ValueError: line 14 of the docstring for coalib.results.SourceRange.SourceRange.__str__ lacks blank after ...: '...test_file'
```

Comment (vijeth-aradhya) [inline coalib/results/SourceRange.py]: I even tried this work around `# doctest: +ELLIPSIS` and `[...]`
Source - http://stackoverflow.com/questions/5812833/can-i-have-an-ellipsis-at-the-beginning-of-the-line-in-a-python-doctest

Comment (Makman2) [inline coalib/results/SourceRange.py]: ah shit yeah maybe it recognizes the leading `...` as the continuation line like it appears in the python interpreter.

Comment (Makman2) [inline coalib/results/SourceRange.py]: haha wait we forgot to use string quotes :3
```
'...test_file'
```
this should work I think :D

Comment (Makman2) [inline coalib/results/SourceRange.py]: haha wait we forgot to use string quotes :3
```
'...test_file'
```
this should work I think :D

Comment (vijeth-aradhya) [inline coalib/results/SourceRange.py]: Um Nope :/ It still doesn't read it as regex
```
Expected:
    '...test_file'
```

File: coalib/results/SourceRange.py [hunk 20]
```diff
@@ -126,6 +126,51 @@ def __json__(self, use_relpath=False):
...
+
+        If the whole line is affected, then just the filname with starting
```
Comment (Makman2) [inline coalib/results/SourceRange.py]: typo: filname -> filename :)

File: coalib/results/SourceRange.py (hunk 19 above)
Comment (Makman2) [inline coalib/results/SourceRange.py]: works like a charm on my system, tested that on your branch^^
```
>>> str(SourceRange.from_values('test_file', None, None, None, None))
'...test_file'
```

Comment (Makman2) [inline coalib/results/SourceRange.py]: > you don't need the `+ELLIPSIS` for this to work, it's active by default

indeed, works for me without ellipsis :)

File: tests/results/SourceRangeTest.py (hunk 18 above)
Comment (Makman2) [inline tests/results/SourceRangeTest.py]: yes cool, do this :)

though what's the defined behaviour for that? If it's undefined behaviour you don't need to test it^^

Comment (vijeth-aradhya) [inline tests/results/SourceRangeTest.py]: It will just print `None` instead of the number. If the case will never arise at all, we don't have to test it. Are we sure these kind of cases do not arise?

Comment (Makman2) [inline tests/results/SourceRangeTest.py]: Can't guarantee that. Though isn't it better to print something like `L3 : L4C5` or maybe just use column number 1 then? --> `L3C1 : L4C5`

Comment (Makman2) [inline tests/results/SourceRangeTest.py]: > ... or maybe just use column number 1 then? --> L3C1 : L4C5

nvm this would be inconsistent when omitting the end-column, as then we would have to read the file to check out which one is the last column^^ We don't want that^^

So then `L3 : L4C5` is better or `L3C5 : L4`

Comment (vijeth-aradhya) [inline tests/results/SourceRangeTest.py]: But, calculating last line and last column would be a problem as you said - _for example_
if `start.line` is not `None` and `end.line` is `None`, then `filename: L3 : ?lastline?`
I'm not sure properly as to what to do when `end.line` or `end.start` is `None` :/

Comment (vijeth-aradhya) [inline tests/results/SourceRangeTest.py]: Technically, we get numbers right - It says `start.line` is interpreted as the first line when it is `None`, but it will be just be 1 and not `None`. And, each `Bear` ( whichever other API or executables we call to get the bear work ) will give **both** column numbers or it just wont give both. And, the same goes with the line.
I think we should stick to these 3 boundary cases. It is very unlikely that the above discussed cases will occur. What do you think?

Comment (Makman2) [inline tests/results/SourceRangeTest.py]: yeah this gets really complicated then... Agreeing :+1: :)

**Comment by vijeth-aradhya** [comment]
I have currently removed the `test_str` method as discussed. I still vouch having though for testing other cases in future.
But yeah, the others should be fine I guess?!

File: coalib/results/SourceRange.py [hunk 21]
```diff
@@ -126,6 +126,41 @@ def __json__(self, use_relpath=False):
...
+        If the whole line is affected, then just the filename with starting
+        line number and ending line number is shown.
+
+        >>> str(SourceRange.from_values('test_file', 1, 1, 2, 1))
+        '...test_file: L1 C1 : L2 C1'
+
+        This is the general case where particular column and line are
+        specified. It shows the starting line and column and ending line
+        and column, with filename in the beginning.
+
```
Comment (sils) [inline coalib/results/SourceRange.py]: why this newline?

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/3394/commits/130e029aff2ff3712daad6eee8658b856378eb11

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/3394/commits/f37a86d0e28368fe386020273b0aa0a1e71a34ca

**Comment by sils** [comment]
https://github.com/coala/coala/pull/3394/commits/c9e2e3e5ec05add023e09f8c4aed0bc05e316362 unack

**Comment by Makman2** [comment]
@vijeth-aradhya We introduce the test function as needed, having an empty/meaningless test function is not good :)

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/3394/commits/d3c8ec930872b9dba576f764902c76fb645f0d60 https://github.com/coala/coala/pull/3394/commits/2614c792f15db6090f8482bb21db7f06b3ba83a0

**Comment by sils** [comment]
@vijeth-aradhya canyou rebase this one last time so we can merge it? :)

File: coalib/results/SourceRange.py [hunk 22]
```diff
@@ -126,6 +126,40 @@ def __json__(self, use_relpath=False):
             _dict['file'] = relpath(_dict['file'])
         return _dict
 
+    def __str__(self):
+        """
+        Creates a string representation of the SourceRange object
```
Comment (Makman2) [inline coalib/results/SourceRange.py]: Missing period at end of sentence

File: coalib/results/SourceRange.py [hunk 23]
```diff
@@ -126,6 +126,40 @@ def __json__(self, use_relpath=False):
...
+        Here are doctest examples of the three different representations
```
Comment (Makman2) [inline coalib/results/SourceRange.py]: I would leave this sentence away, the user can see that examples are following :)

File: coalib/results/SourceRange.py (hunk 22 above)
Comment (Makman2) [inline coalib/results/SourceRange.py]: \+ you could enclose SourceRange into double backticks: `´´SourceRange´´` (I used `´` here because github doesn't allow to escape `), so it gets rendered as code

File: coalib/results/SourceRange.py [hunk 24]
```diff
@@ -126,6 +126,40 @@ def __json__(self, use_relpath=False):
...
+
+        >>> str(SourceRange.from_values('test_file', None, None, None, None))
+        '...test_file'
+
+        If the whole file is affected, then just the filename is shown.
```
Comment (Makman2) [inline coalib/results/SourceRange.py]: This sentence should be located above the example

File: coalib/results/SourceRange.py [hunk 25]
```diff
@@ -126,6 +126,40 @@ def __json__(self, use_relpath=False):
...
+
+        >>> str(SourceRange.from_values('test_file', 1, None, 2, None))
+        '...test_file: L1 : L2'
+
+        If the whole line is affected, then just the filename with starting
+        line number and ending line number is shown.
```
Comment (Makman2) [inline coalib/results/SourceRange.py]: this one should be also above the example

File: coalib/results/SourceRange.py [hunk 26]
```diff
@@ -126,6 +126,40 @@ def __json__(self, use_relpath=False):
...
+
+        >>> str(SourceRange.from_values('test_file', 1, 1, 2, 1))
+        '...test_file: L1 C1 : L2 C1'
+
+        This is the general case where particular column and line are
+        specified. It shows the starting line and column and ending line
+        and column, with filename in the beginning.
```
Comment (Makman2) [inline coalib/results/SourceRange.py]: this one too

File: coalib/results/SourceRange.py [hunk 27]
```diff
@@ -126,6 +126,40 @@ def __json__(self, use_relpath=False):
...
+        """
+
```
Comment (Makman2) [inline coalib/results/SourceRange.py]: up to you, but I would remove this newline :)

**Comment by Makman2** [comment]
Only this commit:
`SourceRange.py: Add __str__ method `
Solves the referenced issue. Please remove all other `Closes xxx` notes from the other commits, or change them to `Related to xxx` :+1:

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/3394/commits/48d19b3c521ab46a2aff8595b5563cd2cde66425 https://github.com/coala/coala/pull/3394/commits/c769a92ba88613932f3b8b881feac37d65ee2b7c https://github.com/coala/coala/pull/3394/commits/225bc9fe74243a4e4d2a34a643fea9ab5439c233

**Comment by sils** [comment]
@rultor merge

**Comment by sils** [comment]
:tada: awesome :)

At the danger of sending you info that you already received: Congrats and welcome to the coalaians! :) You've done your first step to become a coala developer.

Rultor will now automatically release your changes as a prerelease so users can experience the goodness of your changes right now.

We would really appreciate it if you could help us with

- [telling us about your experience](http://coala.io/newform)
- [some code reviews](http://coala.io/reviewing) for other newcomers
- fixing at least [a difficulty/low issue](http://coala.io/low)
- fill our [usability survey](http://coala.io/usability)

If you've done those and tell us about it so we can promote you to a coala developer!

We're all volunteers and we're currently struggling to keep this up. Helping all those newcomers is a lot of work and we do need your help so we can continue this for others!

**Comment by vijeth-aradhya** [comment]
@sils Yeah I have done a couple of basic reviews if that counts. And, this counts as one `difficulty/low` right? :) I've also added `GhcModBear` [PR](https://github.com/coala/coala-bears/pull/1223) just now as well. Would love to be a part of this community :D
I need to fill up the survey form though!

**Comment by sils** [comment]
cool, ping us on gitter when you want to obe promoted :)
